### ISSUE TITLE
Standardize provided dataset label and data members

### ISSUE THREAD

**Issue by jramapuram** [pytorch/vision#475]
This is in related to: https://discuss.pytorch.org/t/solved-simplest-way-to-increment-multiple-dataset-labels-using-the-operator/16809/5

Summary:  There is no consistent way to access members such as the dataset labels across the provided datasets, eg:

MNIST: uses self.train_labels here: https://github.com/pytorch/vision/blob/master/torchvision/datasets/mnist.py#L53
SVHN : uses self.labels here: https://github.com/pytorch/vision/blob/master/torchvision/datasets/svhn.py#L76

This should be standardized across the provided datasets in case for example users want to concat datasets and increase the labels as the datasets increase (see forum summary for more info)

**Comment by fmassa** [comment]
If the user wants to concatenate both datasets, why not use [`torch.utils.data.ConcatDataset`](http://pytorch.org/docs/stable/data.html#torch.utils.data.ConcatDataset)?

**Comment by jramapuram** [comment]
@fmassa : this doesn't have logic to change the labels of the second dataset. Eg:

1. dataset0 :  labels 0-9
2. dataset1 : labels 0-9

`ConcatDataset` here will not change the labels of dataset1. What I want is something like `ConcatDatasetExtendLabels` to ensure that dataset1's labels to be shifted from 10-19

**Comment by fmassa** [comment]
I think you can do this in a simple manner: add a `target_transform` to one of your datasets that shifts the labels, and then use `ConcatDataset`
What do you think?

**Comment by jramapuram** [comment]
Good point! Didn't actually think of this. This is probably the most appropriate way to get access to the underlying members
